src/services/appointment.ts

```diff
@@ -50,4 +52,81 @@ const updateIsAvailable = async (id: string, isAvailable: boolean) => {
   return afterUpdate;
 };
 
-export { getAppointmentsPerDateService, getAppointmentById, updateIsAvailable };
+function generateAppointments(
+  start: string,
+  end: string,
+  timeRanges: string[],
+  therapistId:number,
+): AppointmentType[] {
+  const startDate = moment(start, 'YYYY-MM-DD');
+  const endDate = moment(end, 'YYYY-MM-DD');
+  const appointments: AppointmentType[] = [];
+
+  const availableTimeRanges: Duration[][] = timeRanges.map((range) => {
+    const [startTime, endTime] = range.split('-');
+    const startMoment = moment(startTime, 'HH:mm');
+    const endMoment = moment(endTime, 'HH:mm');
+
+    if (endMoment.isBefore(startMoment)) {
+      return [
+        moment.duration(startMoment.diff(moment().startOf('day'))),
+        moment.duration(endMoment.diff(moment().startOf('day').add(1, 'day'))),
+      ];
+    }
+
+    return [moment.duration(startMoment.diff(moment().startOf('day'))), moment.duration(endMoment.diff(moment().startOf('day')))];
+  });
+
+  const currentDate = startDate.clone().startOf('day');
+
+  while (currentDate.isSameOrBefore(endDate, 'day')) {
+    const availableRanges = availableTimeRanges.map((range) => {
+      const startTime = currentDate.clone().startOf('day').add(range[0]);
+      const endTime = currentDate.clone().startOf('day').add(range[1]);
+
+      return { startTime, endTime };
+    });
+
+    const hourRange = moment.duration(1, 'hour');
+
+    availableRanges.forEach(({ startTime, endTime }) => {
+      const currentTime = startTime.clone();
+
+      while (currentTime.isBefore(endTime)) {
+        appointments.push({
+          id: 21 + appointments.length + 1,

```

Comment:
for now, until I solve the date problem, I don't want to think about the id problem, I will fix it later